### DISCUSSION TITLE
PnP V4 Version Upgrade

### DISCUSSION THREAD

**Discussion by aavi92** [microsoft-search/pnp-modern-search#2604] [Q&A]
Hi we are currently on PnP Modern Search V4 4.3.0 and wanted to upgrade to 4.8.0 i understand we need to uninstall the older version to be able to upgrade to newer one and few sites in our tenant were configured using 4.3.0  is there a way i can upgrade to 4.8.0 with out impacting the current search configurations on sites. 
Thanks

**Comment by ElephantLathe** [comment]
In a similar position, we use PNP Search web parts for all sorts of content surfacing in multiple sites, not just for general Search, previous v4 versions have just updated in place via the app store with no issues, might be misunderstanding the problems, but will we need to uninstall and reinstall PNP Search and reconfigure every single PNP Search web part we've added? Is there a way to still do that in place upgrade without doing that? 

I've looked at this Twitter thread, but I'm don't really understand the issue https://twitter.com/mikaelsvenson/status/1587500092671627265

**Reply by ElephantLathe** [reply]
Tested this out in a test tenancy, older version of PNP search v4.61 deployed globally in the tenancy, and a few search pages set up with filters, search results and search box. I removed the existing web part from the central SharePoint App store, checked the pages, nothing displayed.

Uploaded v4.8.0 deployed globally, approved APIs it asked for and the pages all worked afterwards. So although it's not the same name sppkg package being uploaded over the top of the previous, the one that is uploaded must have the same references for existing web parts on pages to work, so all looking ok to upgrade in production for us.

**Reply by wobba** [reply]
There is no need to reconfigure the web parts on pages - those id's are all the same. It's an internal feature id which we changed back to the "correct" setting. 4.3 to 4.8 I believe should be just fine. It's those having 4.6/4.7 which are affected I believe.
